docs/modules/SDDs/pages/0022-managed-services-in-cluster.adoc

```diff
@@ -0,0 +1,111 @@
+
+This doesn't include services offered by the cloud.
+
+=== Goals
+
+* Offer application services to applications running in or outside of the cluster
+* Maintenance processes to maintain these application services, making them a managed service
+* Definition of how the end user application instantiates and consumes an application service
+
+=== Non-Goals
+
+* Services offered by the cloud are not part of this SDD
+* All plumbing and system services are discussed in other SDDs
+
+== Design Proposal
+
+It is considered that all application services are managed by a specialized Kubernetes Operator, which brings the domain knowledge about the specific service. This Kubernetes Operator will be installed and configured by a Commodore Component. The Commodore Component will be responsible for:
+
+* Installing the Kubernetes Operator
+* Pre-configuring the Kubernetes Operator with best practices, if applicable
+* Deploy additional objects like monitoring definition, backup definition or any other object needed to make the application service a managed service
+
+When the Kubernetes Operator is installed and operational, the user of the platform requests an instance of the application service by creating an Operator specific object. The job of the Kubernetes Operator is to instantiate the service and provide the information to the user for actually connecting to the service.
+
+This SDD doesn't define how this service instance object is handled and managed. It could come together with the application deployment objects of the customer but also being stored in a separate GitOps repository just for these objects.
+
+A Kubernets Operator could also be responsible to manage different kinds of application services, it isn't strictly necessary that each Operator is only responsible for one service.
+
+Each provided service will be documented in the Commodore Component which installs the Kubernetes Operator for the service.
+
+=== User Stories
+
+==== Kafka with Strimzi Operator - Application in Cluster
+
+_As the user of the platform I want to use Apache Kafka for my application runnign in the same Kubernetes cluster as a managed service._
+
+* Project Syn Commodore Component `component-kafka-strimzi` deploys Strimzi to the cluster and configures it according to best practices.
+* The user of the platform creates a `Kafka` object and puts in parameters as needed. The object is stored at the same place as the other application deployment objects and also deployed to the cluster with the same mechanisms (f.e. GitOps).
+* The Strimzi Kubernetes Operator deploys Kafka and creates a secret with the connection parameters for the users application to consume.
+* Via the monitoring and backup configuration deployed by the Commodore Component the application service is automatically monitored and backed up.
+* For keeping the Strimzi Kubernetes Operator up-to-date the Project Syn mechanisms are used.
```

Comment:
Cross link to https://github.com/projectsyn/documentation/blob/master/docs/modules/SDDs/pages/0017-maintenance-with-renovate.adoc ?